tests/Mado/QueryBundle/Component/Sherlock/SherlockTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Mado\QueryBundle\Tests\Objects;
+
+use Mado\QueryBundle\Component\Sherlock\Sherlock;
+use PHPUnit\Framework\TestCase;
+
+class SherlockTest extends TestCase
+{
+    public function setUp()
+    {
+        $this->manager = \Doctrine\ORM\EntityManager::create(array(
+            'driver' => 'pdo_sqlite',
+            'path' => __DIR__ . '/../../data/db.sqlite',
+        ),
+        \Doctrine\ORM\Tools\Setup::createAnnotationMetadataConfiguration(
+            array(__DIR__),
+            true
+        ));
+
+        $this->sherlock = new Sherlock(
+            $this->manager
+        );
+    }
+
+    public function testShouldProvideAvailableOperatorsForASpecificEntity()
+    {
+        $this->assertEquals(
+            [
+                'fields' => [
+                    'id' => [
+                        'eq',
+                        'neq',
+                        'gt',
+                        'gte',
+                        'lt',
+                        'lte',
+                    ],
+                    'username' => [
+                        'startswith',
+                        'contains',
+                        'notcontains',
+                        'endswith',
+                    ],
+                ],
+                'relations' => [
+                    [
+                        'middle' => 'mado.querybundle.tests.objects.middleentity',
+                        'tooMany' => 'mado.querybundle.tests.objects.toomany',
+                    ]
+                ]
+            ],
+            $this->sherlock->getOpList("mado.querybundle.tests.objects.startingentity")
+        );
+    }
+
+    public function testExtractJustFieldTypeAndNotOperatorList()
+    {
+        $this->assertEquals(
```

Comment:
@marcocasadio whenever will be necessary just field type there will be this method!